### PR TITLE
Bob 64 update death screen

### PR DESCRIPTION
**Pull request by zicons**

### CODE REVIEW

**Review by FurkanUses — CHANGES_REQUESTED** [review]
I can't see the curser in the Deathscreen and also then I startet a new game I spawn out side the map (but I think the second problem needs to be fixed some where else).

File: Assets/Scripts/ResourceManager.cs [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 public static class ResourceManager
 {
+    private static List<Resource> Inventory = new List<Resource>();
```
[the remark this answers is not in the document]
Comment (SomethingWittyWithATechAspect) [reply Assets/Scripts/ResourceManager.cs:9]: create class Item with quantity, name, (maybe sprite)
resource should extend this item
Inventory can be list of items

depending on how we want to have the player manage the inventory we might want list or set
list when either items can be stacked and inventory can have mutltiple stacks of the same item or when no stacks and each item fills a single slot. 

Set if unlimeted stacks per item

**Review by SomethingWittyWithATechAspect — CHANGES_REQUESTED** [review]

**Comment by SomethingWittyWithATechAspect** [comment]
> I can't see the curser in the Deathscreen and also then I startet a new game I spawn out side the map (but I think the second problem needs to be fixed some where else).

im not spawning out of map
but ditto to the cursor thing (only in build) - easiest fix is to add crosshair to deathscreen or unhide cursor on death
in the first case we should also add crosshair to menu and hide the cursor at the very beginning

File: Assets/Editor/Tests/BookMenuTest.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    // A UnityTest behaves like a coroutine in Play Mode. In Edit Mode you can use
+    // `yield return null;` to skip a frame.
+    [UnityTest]
+    public IEnumerator BookMenuTestWithEnumeratorPasses()
+    {
+        // Use the Assert class to test conditions.
+        // Use yield to skip a frame.
+        yield return null;
+    }
+}
```
[the remark this answers is not in the document]
Comment (zicons) [reply Assets/Editor/Tests/BookMenuTest.cs:25]: was just scafolding..  i will remove it.

File: Assets/Tests/PlayMode/BookUITests.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            _clicked = true;
+        }
+
+        void OnSceneLoaded(Scene scene, LoadSceneMode mode)
+        {
+            _sceneLoaded = true;
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (zicons) [reply Assets/Tests/PlayMode/BookUITests.cs:96]: done

File: Assets/Scripts/ResourceManager.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (zicons) [reply Assets/Scripts/ResourceManager.cs:9]: what is the difference between Item and resource?

[the remark this answers is not in the document]
Comment (zicons) [reply Assets/Scripts/ResourceManager.cs:9]: changed everything to inventory.

File: Assets/Scripts/GameController.cs [hunk 4]
```diff
@@ -1,18 +1,47 @@
...
+                // mark root as DontDestroyOnLoad();
+                DontDestroyOnLoad(_instance.gameObject);
+            }
+
+            return _instance;
+        }
+    }
+
+    //version of the one below with one parameter to be able to connect UnityEvents
+    public void AddOnePoint()
```
[the remark this answers is not in the document]
Comment (zicons) [reply Assets/Scripts/GameController.cs:29]: ok

File: Assets/Scripts/Attributes/HealthSystemAttribute.cs [hunk 5]
```diff
@@ -48,8 +50,11 @@ public void ModifyHealth(int amount)
         //DEAD
         if (health <= 0)
         {
-            ui.AddOnePoint();
-            Destroy(gameObject);
+            gc.AddOnePoint();
```
[the remark this answers is not in the document]
Comment (zicons) [reply Assets/Scripts/Attributes/HealthSystemAttribute.cs:53]: ok

**Review by FurkanUses — APPROVED** [review]

**Review by SomethingWittyWithATechAspect — APPROVED** [review]

**Comment by SomethingWittyWithATechAspect** [comment]
I meant to do some more refinement in resourceMangement is needed but i will work this out and create an issue/task

**Pull request merged**